### ISSUE TITLE
Pretty printer failing

### ISSUE THREAD

**Issue by marcelotto** [michalmuskala/jason#49]
The following example shows a case where the new pretty printer not only fails to pretty print, but also changes the data. Note the added whitespace to the URIs.
```elixir
iex> Jason.encode!([%{"@id" => "http://a/b", "http://a/c" => [%{"@id" => "http://a/d"}]}], pretty: true)
[
  {
    "@id":"http: //a/b","http: //a/c":[{"@id":"http: //a/d"}]}]
```

**Comment by michalmuskala** [comment]
The minimal case is `Jason.encode!(["//"], pretty: true)` I'll look into it.

**Comment by michalmuskala** [comment]
The fix is published on hex in 1.1.1

**Comment by marcelotto** [comment]
Thank you very much for the fix, this library in general and in particular the new pretty printing feature, which allowed me to cross a highly-wanted feature for [JSON-LD.ex](https://github.com/marcelotto/jsonld-ex) from my TODO list with the least possible effort.
